tests/core/test_registry_contract.py

```diff
@@ -0,0 +1,129 @@
+import logging
+
+import pytest
+
+from eth_tester.exceptions import TransactionFailed
+
+PACKAGE_ID = b'\xd0Y\xe8\xa6\xeaZ\x8b\xbf\x8d\xd0\x97\xa7\xb8\x92#\x16\xdc\xb7\xf0$\xe8"\x0bV\xd3\xc9\xe7\x18\x8ajv@'  # noqa: E501
+V1_RELEASE_ID = b"Y^&\xf1\xb2${\xac\xc5~2\x80\x7f\x80Y\xe0\xb0\x83}\xd9\xc4~iF\x99A\x96\xbd)\xc9\xca\x97"  # noqa: E501
+V2_RELEASE_ID = b"\x04Y,\xb9\xce\xd5A>\x1b\t\xe8{\x08\x9b\xf6\x96\xa0^\xfbv\xee\x87\xc8\xc4\x12Yc_\xacm\x93\x8a"  # noqa: E501
+
+logging.getLogger("evm").setLevel(logging.INFO)
+
+
+@pytest.fixture
+def registry(deployer):
+    pkg = deployer.deploy("registry")
+    return pkg.deployments.get_contract_instance("registry")
+
+
+def test_registry_init(registry, w3):
+    assert registry.functions.owner().call() == w3.eth.accounts[0]
+
+
+def test_registry_generate_release_id(registry):
+    release_id = registry.functions.generate_release_id(b"package", b"1.0.0").call()
+    assert release_id == V1_RELEASE_ID
+
+
+def test_registry_release(registry):
+    release_id = registry.functions.generate_release_id(b"package", b"1.0.0").call()
+    assert release_id == V1_RELEASE_ID
+
+
+def test_registry_get_package_name_raises_exception_if_package_doesnt_exist(registry):
+    with pytest.raises(TransactionFailed):
+        registry.functions.get_package_name(PACKAGE_ID).call()
+
+
+def test_registry_get_package_name(registry):
+    registry.functions.release(b"package", b"1.0.0", b"google.com").transact()
+    package_name = registry.functions.get_package_name(PACKAGE_ID).call()
+    assert package_name[:7] == b"package"
+
+
+def test_registry_logs_release_event(registry, w3):
+    tx_hash = registry.functions.release(b"package", b"1.0.0", b"google.com").transact()
+    receipt = w3.eth.waitForTransactionReceipt(tx_hash)
+    logs = registry.events.Release().processReceipt(receipt)
+    assert logs[0]["args"]["_package"][:7] == b"package"
```

Comment:
Same here, another pattern we want to make easier (as suggested by @fubuloubu )

Here are some ideas for starting points.

```python
# test specific log
log = ...
assert Log(registry.events.Release, "package", "1.0.0", "google.com").match(log)

# or maybe test it is just present in a receipt
receipt = ...
assert Log(registry.events.Release, "package", "1.0.0", "google.com").is_present(receipt)

# or test that there is no log entry
assert Log(registry.events.Release, "package", "1.0.0", "google.com").not_present(receipt)
```

The `Log` constructor would need to do some magic introspecting the `ContractEvent` object returned by `registry.events.Receipt` but I think all the necessary information is there.